src/string/repeated-substring-pattern.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Repeated Substring Pattern
+ *
+ * Given a non-empty string check if it can be constructed by taking a substring of it
+ * and appending multiple copies of the substring together.
+ * You may assume the given string consists of lowercase English letters only and its length will not exceed 10000.
+ *
+ * Example 1:
+ *
+ * Input: "abab"
+ * Output: True
+ *
+ * Explanation: It's the substring "ab" twice.
+ *
+ * Example 2:
+ *
+ * Input: "aba"
+ * Output: False
+ *
+ * Example 3:
+ *
+ * Input: "abcabcabcabc"
+ * Output: True
+ *
+ * Explanation: It's the substring "abc" four times. (And the substring "abcabc" twice.)
+ */
+
+/**
+ * Smart Solution
+ *
+ * @param {string} s
+ * @return {boolean}
+ */
+const repeatedSubstringPattern = s => {
+  const str = s + s;
+  return str.substring(1, str.length - 1).includes(s);
+};
+
+/**
+ * Recursive Solution
+ *
+ * @param {string} s
+ * @return {boolean}
+ */
+const repeatedSubstringPattern = s => {
+  const canRepeat = (s, rest) => {
```

Comment:
If you do not intend to use variables from outside `canRepeat`'s scope, declare it outside to avoid building it each time you call `repeatedSubstringPattern`.